leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/CaliforniumEndpointsManager.java

```diff
@@ -0,0 +1,198 @@
+        }
+
+        // Add new endpoint
+        coapServer.addEndpoint(currentEndpoint);
+
+        // Start endpoint if needed
+        if (started) {
+            coapServer.start();
+            try {
+                currentEndpoint.start();
+                LOG.info("New endpoint created for server {} at {}", serverInfo.serverUri, currentEndpoint.getUri());
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to start endpoint", e);
+            }
+        }
+        return server;
+    }
+
+    private Builder cloneDtlsConfigBuilder(Builder builder) {
+        // TODO we must see if we can add this to californium
+        DtlsConnectorConfig incompleteConfig = builder.getIncompleteConfig();
+        Builder newBuilder = new Builder();
+
+        newBuilder.setAddress(incompleteConfig.getAddress());
+        newBuilder.setMaxConnections(incompleteConfig.getMaxConnections());
+        newBuilder.setStaleConnectionThreshold(incompleteConfig.getStaleConnectionThreshold());
+        newBuilder.setConnectionThreadCount(incompleteConfig.getConnectionThreadCount());
+
+        if (incompleteConfig.isAddressReuseEnabled() != null)
+            newBuilder.setEnableAddressReuse(incompleteConfig.isAddressReuseEnabled());
+        if (incompleteConfig.isClientAuthenticationRequired() != null)
+            newBuilder.setClientAuthenticationRequired(incompleteConfig.isClientAuthenticationRequired());
+        if (incompleteConfig.isEarlyStopRetransmission() != null)
+            newBuilder.setEarlyStopRetransmission(incompleteConfig.isEarlyStopRetransmission());
+        if (incompleteConfig.getMaxFragmentLengthCode() != null)
+            newBuilder.setMaxFragmentLengthCode(incompleteConfig.getMaxFragmentLengthCode());
+        if (incompleteConfig.getMaxRetransmissions() != null)
+            newBuilder.setMaxRetransmissions(incompleteConfig.getMaxRetransmissions());
+        if (incompleteConfig.getOutboundMessageBufferSize() != null)
+            newBuilder.setOutboundMessageBufferSize(incompleteConfig.getOutboundMessageBufferSize());
+        if (incompleteConfig.getRetransmissionTimeout() != null)
+            newBuilder.setRetransmissionTimeout(incompleteConfig.getRetransmissionTimeout());
+        if (incompleteConfig.getServerNameResolver() != null)
+            newBuilder.setServerNameResolver(incompleteConfig.getServerNameResolver());
+        if (incompleteConfig.getTrustStore() != null)
+            newBuilder.setTrustStore(incompleteConfig.getTrustStore());
+
+        // TODO we should probably clean cipherSuite depending on which kind of endpoint(psk,rpk,x509) we will create
+        // in case users choose specific cipher
+
+        return newBuilder;
+    }
+
+    @Override
+    public synchronized Collection<Server> createEndpoints(Collection<? extends ServerInfo> serverInfo) {
+        if (serverInfo == null || serverInfo.isEmpty())
+            return null;
```

Comment:
wouldn't be safer to return an empty collection instead of null?

Reply:
I will do that.